named_arrays/_functions/functions.py

```diff
@@ -45,6 +45,57 @@ def outputs(self) -> na.AbstractArray:
 
     __named_array_priority__: ClassVar[float] = 100 * na.AbstractVectorArray.__named_array_priority__
 
+    @property
+    def axes_center(self) -> tuple(str):
+        """
+        Return keys corresponding to all input axes representing bin centers
+        """
+        axes_center = tuple()
+        input_shape = self.inputs.shape
+        output_shape = self.outputs.shape
+        all_axes = set(input_shape | output_shape)
+
+        for axis in all_axes:
+            if axis in output_shape:
+                if axis in input_shape:
+                    if input_shape[axis] == output_shape[axis]:
+                        axes_center += (axis,)
+                    elif input_shape[axis] != output_shape[axis] + 1:
+                        raise ValueError(
+                            'Output axis dimension must either match Input axis dimension (representing'
+                            ' bin centers, or exceed by one (representing bin vertices).'
+                        )
+                else:
+                    axes_center += (axis,)
+
+            if axis in input_shape:
+                if axis not in output_shape:
+                    axes_center += (axis,)
```

Comment:
I wonder if we could make this logic simpler by broadcasting the input and output arrays together first.

Reply:
does this do the same thing?

```suggestion
        axes_common = set(output_shape) & set(input_shape)
        for axis in all_axes:
            if axis in axes_common:
                if input_shape[axis] == output_shape[axis]:
                    axes_center += (axis,)
                elif input_shape[axis] != output_shape[axis] + 1:
                    raise ValueError(
                        'Output axis dimension must either match Input axis dimension (representing'
                        ' bin centers, or exceed by one (representing bin vertices).'
                    )
            else:
                axes_center += (axis,)
```

Reply:
I think so.  Can we also just say, if axis in output_shape or input_shape: instead of defining axes_common?